tests/test_designmatrix_noisepars.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+from pint.models import get_model_and_toas
+from pinttestdata import datadir
+
+
+@pytest.fixture()
+def model_and_toas():
+    parfile = datadir / "J1713+0747_small.gls.par"
+    timfile = datadir / "J1713+0747_small.tim"
+    return get_model_and_toas(parfile, timfile)
+
+
+def test_designmatrix_no_free_noise_param(model_and_toas):
+    model, toas = model_and_toas
+
+    # This should work.
+    M = model.designmatrix(toas)
```

Comment:
Are there any properties M should have that you can `assert` here? Or if you just want the function call to not raise an exception, maybe say so in the function name? (Also you don't have to assign the return value to anything if you never use it.)

Reply:
I have added new assertions to test for the shape of the design matrix.